### ISSUE TITLE
site proof job is failing on master builds

### ISSUE THREAD

**Issue by skriss** [projectcontour/contour#2624]
e.g. https://travis-ci.com/github/projectcontour/contour/jobs/353156833:

```
- ./_site/docs/v1.2.1/api-reference.html
  *  External link https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#objectmeta-v1-meta failed: 404 No error
- ./_site/docs/v1.5.1/api-reference.html
  *  External link https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#loadbalancerstatus-v1-core failed: 404 No error
- ./_site/guides/proxy-proto/index.html
  *  External link http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt failed: got a time out (response code 0)
```

Looks like there are a few things we need to do to get this back to green:
- [x] update https://github.com/projectcontour/contour/blob/master/site/_data/refdocs/config.json#L16 to use a v1.18 link since the v1.13 docs seem to no longer be hosted (https://github.com/projectcontour/contour/pull/2625)
- [ ] either (a) update all the old versioned docs links to use a v1.18 link, or (b) exclude old versioned docs from the proofer (#2627)
- [ ] fix the problem with [this link](https://github.com/projectcontour/contour/blame/master/site/_guides/proxy-proto.md#L55) (#2627)

**Comment by stevesloka** [comment]
We should also figure out why the CI job doesn't pick these errors up.

**Comment by stevesloka** [comment]
The CI job doesn't pick up changes, see your commit on #2625, site-proof was skipped: https://travis-ci.com/github/projectcontour/contour/jobs/353230458#L288

**Comment by skriss** [comment]
Looks like it's because of how Travis does PR builds, where it merges the PR changes into master before building, which isn't compatible with the detect-changes script. I'll look at updating the script to work for PRs, but we won't want to merge that until we get the master builds green, otherwise all PR builds are going to start failing.

**Comment by skriss** [comment]
@stevesloka found some more related issues here - it looks like we also need to fix #2032 before getting this running for PRs, so the fact that it's not running currently is more of a feature than a bug :)

**Comment by skriss** [comment]
Not sure why [this link to haproxy.org](https://github.com/projectcontour/contour/blame/master/site/_guides/proxy-proto.md#L55) is timing out - it passes the check on my machine.
